### PR TITLE
[3.6] bpo-29581: Make __init_subclass__ work on ABCMeta type classes

### PR DESCRIPTION
**Pull request by Soares**
I'm not entirely sure that I used the cherry_pick tool correctly, please make sure that I didn't muck anything up.

I'm also planning to backport PR #678, which depends on this backport (of PR #527). I assume that I shouldn't begin that backport until this one is accepted, but let me know if I can begin cherry-picking #678 now.

### CODE REVIEW

**Comment by Mariatta** [comment]
Sorry with the confusion of how to use cherry_picker.py.

I suppose you were following the instruction [here](http://cpython-devguide.readthedocs.io/committing.html?highlight=backport#backporting-changes-to-python-3-6-or-older-version), which mentions about merging with `master` first. This section is not up to date, and 
there is an outstanding [issue](https://github.com/python/devguide/issues/173) in the devguide about it. 

Once you have cherry_picker.py you'll only need to do `python -m cherry_picker <sha1> 3.6`.
In case there is a merge conflict, you'll need to resolve it manually (I use Resolve Conflict from within Pycharm), and then do `python -m cherry_picker --continue`

Hope this helps a bit :)

If you want, I can do the backport for you later today.

**Comment by Mariatta** [comment]
Nvm, I changed the base branch to `3.6` instead of `master` 😅  Looks better now

**Comment by Soares** [comment]
Bump. What do I need to do to cause this backport to happen?

**Comment by Mariatta** [comment]
Sorry this fell off my radar.
Can you please resolve the conflict? After that this is good to go. Thanks :)

**Comment by Soares** [comment]
Conflict resolved. Thanks!

**Comment by Mariatta** [comment]
Travis build failed.. :thinking:

File: Misc/NEWS [hunk 1]
```diff
@@ -45,6 +45,9 @@ Core and Builtins
 Library
 -------
 
+- bpo-29581: ABCMeta.__new__ now accepts **kwargs, allowing abstract base
```
Comment (Mariatta) [inline Misc/NEWS]: Instead of
```
**kwargs
```
, can you try 
```
``kwargs``
```
That might fix it for travis.

**Review by Mariatta — APPROVED** [review]

**Comment by Mariatta** [comment]
Thanks :tada:

**Pull request merged**